camerainfo.py

```diff
@@ -21,13 +35,12 @@ def _on_exit():
     # Declare a subscriber on the 'rt/camera/info' topic and print each message
     # with the CameraInfo schema.
     sub = session.declare_subscriber(
-        'rt/camera/info',
-        lambda msg: print(CameraInfo.deserialize(msg.value.payload)))
+        'rt/camera/info', camerainfo_listener)
 
     # The declare_subscriber runs asynchronously, so we need to block the main
-    # thread to keep the program running.  We use signal.pause() to do this
+    # thread to keep the program running.  We use time.sleep() to do this
     # but an application could have its main control loop here instead.
-    signal.pause()
+    time.sleep(args.time)
```

Comment:
This doesn't adequately replace the previous signal handler.  The idea is to block here until control-c not to add timer and additional parameter.

If signal is not portable and Python doesn't provide better than sleep then we should have a sleep loop.

```
while True:
    time.sleep(1)
```